### ISSUE TITLE
Too large mse value benchmarking reproduce

### ISSUE THREAD

**Issue by 5taku** [SalesforceAIResearch/uni2ts#184]
**Describe the bug**

I tried to reproduce the moirai benchmark, but the mse value was too large.
I try to ETTh2 dataset , cl = 512 , pl = 96 , patch size = 32

I wanna reproduce 
[Unified Training of Universal Time Series Forecasting Transformers paper](https://arxiv.org/pdf/2402.02592)
table 22 . Full results of long sequence forecasting experiments - zero-shot 

![Image](https://github.com/user-attachments/assets/f5e9ca66-4819-4a5e-ac7d-8c8b205f562d)

**To Reproduce**
Please provide a code snippet of a minimal reproducible example for the error. 

reference code : https://github.com/SalesforceAIResearch/uni2ts/blob/main/project/moirai-1/eval/lsf_small.sh
using main branch 'cli' code

```sh
for pl in 96 192 336 720; do
  python -m cli.eval \
    run_name=lsf_eval \
    model=moirai_1.1_R_small \
    model.patch_size=32 \
    model.context_length=512 \
    data=lsf_test \
    data.dataset_name=ETTh2 \
    data.mode=M \
    data.prediction_length=$pl
done
```

**Expected behavior**

A much lower mse value

**Error message or code output**
Paste the complete error message, including stack trace, or the undesired output that the above snippet produces.
```shell
sh moirai-1/eval/lsf_small.sh 
xxx/outputs/eval/lsf_test/ETTh2/M/prediction_length=96/lsf_eval
[2025-02-13 08:31:48,150][datasets][INFO] - PyTorch version 2.4.1 available.
[2025-02-13 08:31:48,155][datasets][INFO] - JAX version 0.5.0 available.
xxx/python3.10/site-packages/uni2ts/common/env.py:43: UserWarning: Failed to load .env file.
  warnings.warn("Failed to load .env file.")
2785it [00:16, 171.18it/s]
         MSE[mean]  MSE[0.5]  MAE[0.5]  MASE[0.5]  MAPE[0.5]  sMAPE[0.5]       MSIS  RMSE[mean]  NRMSE[mean]   ND[0.5]  mean_weighted_sum_quantile_loss
None  16060.988617   0.30669  0.341384   1.156246   1.421138    0.531467  10.918823  126.731956    93.031024  0.250602                         0.203975
```

**Environment**
* Operating system: ubuntu 
* Python version: Python 3.10.12 (main, Jul 29 2024, 16:56:48) [GCC 11.4.0] on linux
* PyTorch version: '2.4.1+cu121'
* uni2ts version (have you pulled the latest version?) main branch code

**Comment by chenghaoliu89** [comment]
Hi @5taku  can you try moirai_1.0_R_small instead?

**Comment by 5taku** [comment]
> Hi [@5taku](https://github.com/5taku) can you try moirai_1.0_R_small instead?

ok I try to moirai_1.0_R_small 

```sh
for pl in 96 192 336 720; do
  python -m cli.eval \
    run_name=lsf_eval \
    model=moirai_1.0_R_small \
    model.patch_size=32 \
    model.context_length=512 \
    data=lsf_test \
    data.dataset_name=ETTh2 \
    data.mode=M \
    data.prediction_length=$pl
done
```

```
sh moirai-1/eval/lsf_small.sh 
xxx/outputs/eval/lsf_test/ETTh2/M/prediction_length=96/lsf_eval
[2025-02-13 12:37:02,289][datasets][INFO] - PyTorch version 2.4.1 available.
[2025-02-13 12:37:02,291][datasets][INFO] - JAX version 0.5.0 available.
2785it [00:13, 209.04it/s]
      MSE[mean]  MSE[0.5]  MAE[0.5]  MASE[0.5]  MAPE[0.5]  sMAPE[0.5]       MSIS  RMSE[mean]  NRMSE[mean]   ND[0.5]  mean_weighted_sum_quantile_loss
None   1.897266  0.295512  0.334898    1.13831   1.374598    0.525818  10.124692    1.377413     1.011127  0.245841                         0.198997
xxx/outputs/eval/lsf_test/ETTh2/M/prediction_length=192/lsf_eval
[2025-02-13 12:38:45,701][datasets][INFO] - PyTorch version 2.4.1 available.
[2025-02-13 12:38:45,710][datasets][INFO] - JAX version 0.5.0 available.
2689it [00:14, 184.40it/s]
        MSE[mean]  MSE[0.5]  MAE[0.5]  MASE[0.5]  MAPE[0.5]  sMAPE[0.5]       MSIS  RMSE[mean]  NRMSE[mean]  ND[0.5]  mean_weighted_sum_quantile_loss
None  9775.985475  0.359556  0.376821   1.296728   1.544966    0.580752  11.311161   98.873583     72.63972  0.27684                         0.224308
xxx/outputs/eval/lsf_test/ETTh2/M/prediction_length=336/lsf_eval
[2025-02-13 12:40:29,922][datasets][INFO] - PyTorch version 2.4.1 available.
[2025-02-13 12:40:29,924][datasets][INFO] - JAX version 0.5.0 available.
2545it [00:16, 152.60it/s]
        MSE[mean]  MSE[0.5]  MAE[0.5]  MASE[0.5]  MAPE[0.5]  sMAPE[0.5]       MSIS  RMSE[mean]  NRMSE[mean]   ND[0.5]  mean_weighted_sum_quantile_loss
None  2328.731136  0.383805    0.4016   1.411588   1.756084    0.630327  12.184624   48.256928    35.621106  0.296443                          0.24036
```

Although it is much smaller than the results of the previous model, I think it's still quite large. Is this value correct?

**Comment by chenghaoliu89** [comment]
1.  you can treat MSE[0.5] MAE[0.5] as the final results, I think your results are slightly worse than the paper reported
2. since the forecasting values are randomly sampled from prediction distribution, there is no guarantee every time the results are exactly the same.
3.  we make some changes after releasing the reproducing scripts (both code and model). If you want to reproduce the same number in our paper, you may need to rollback to that exp reproducing commit.

**Comment by 5taku** [comment]
> 1. you can treat MSE[0.5] MAE[0.5] as the final results, I think your results are slightly worse than the paper reported
> 2. since the forecasting values are randomly sampled from prediction distribution, there is no guarantee every time the results are exactly the same.
> 3. we make some changes after releasing the reproducing scripts (both code and model). If you want to reproduce the same number in our paper, you may need to rollback to that exp reproducing commit.

I fully understand the content. 
Thank you for your response. 😊

**Comment by chenghaoliu89** [comment]
Hi @5taku , after checking you cmd, I think you made some changes. Could you try the original configuration first? model.patch_size=64, model.context_length=5000

**Comment by 5taku** [comment]
> Hi [@5taku](https://github.com/5taku) , after checking you cmd, I think you made some changes. Could you try the original configuration first? model.patch_size=64, model.context_length=5000

right
I change contect length to 512 
ok 
I reproduce context length 5000

but 

```
I wanna reproduce
[Unified Training of Universal Time Series Forecasting Transformers paper](https://arxiv.org/pdf/2402.02592)
table 22 . Full results of long sequence forecasting experiments - zero-shot
```

table 22 result's context length is 512 right?

**Comment by chenghaoliu89** [comment]
> > Hi [@5taku](https://github.com/5taku) , after checking you cmd, I think you made some changes. Could you try the original configuration first? model.patch_size=64, model.context_length=5000
> 
> right I change contect length to 512 ok I reproduce context length 5000
> 
> but
> 
> ```
> I wanna reproduce
> [Unified Training of Universal Time Series Forecasting Transformers paper](https://arxiv.org/pdf/2402.02592)
> table 22 . Full results of long sequence forecasting experiments - zero-shot
> ```
> 
> table 22 result's context length is 512 right?

No, the context length and patch size for each dataset in LSF is in our reproducing scripts. We treat them as hyperparameter and use validation set to select them.

**Comment by 5taku** [comment]
> > > Hi [@5taku](https://github.com/5taku) , after checking you cmd, I think you made some changes. Could you try the original configuration first? model.patch_size=64, model.context_length=5000
> > 
> > 
> > right I change contect length to 512 ok I reproduce context length 5000
> > but
> > ```
> > I wanna reproduce
> > [Unified Training of Universal Time Series Forecasting Transformers paper](https://arxiv.org/pdf/2402.02592)
> > table 22 . Full results of long sequence forecasting experiments - zero-shot
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > table 22 result's context length is 512 right?
> 
> No, the context length and patch size for each dataset in LSF is in our reproducing scripts. We treat them as hyperparameter and use validation set to select them.

I understand that as a result of selecting the best length of each dataset through the validation set and applying it respectively. 
Thank you for your answer. 😄
